test/tap/tests/test_admin_stats-t.cpp

```diff
@@ -0,0 +1,180 @@
+/**
+ * @file test_admin_stats-t.cpp
+ * @brief This tests the Statistics module and its lookup code and tables.
+ * @date 2021-10-28
+ */
+
+#include <algorithm>
+#include <string>
+#include <stdio.h>
+#include <unistd.h>
+#include <vector>
+
+#include <mysql.h>
+#include <mysql/mysqld_error.h>
+
+#include "tap.h"
+#include "command_line.h"
+#include "utils.h"
+
+using std::string;
+
+int main(int argc, char** argv) {
+	CommandLine cl;
+
+	if (cl.getEnv()) {
+		diag("Failed to get the required environmental variables.");
+		return -1;
+	}
+
+    /** @brief Minimum number of distinct variable_name strings in the history_mysql_status_variables_lookup table */
+    const int min_distinct_variable_names = 50;
+
+	plan(5);
+
+	MYSQL* proxysql_admin = mysql_init(NULL);
+
+	// Initialize connections
+	if (!proxysql_admin) {
+		fprintf(stderr, "File %s, line %d, Error: %s\n", __FILE__, __LINE__, mysql_error(proxysql_admin));
+		return -1;
+	}
+
+	// Connnect to local proxysql
+	if (!mysql_real_connect(proxysql_admin, cl.host, cl.admin_username, cl.admin_password, NULL, cl.admin_port, NULL, 0)) {
+		fprintf(stderr, "File %s, line %d, Error: %s\n", __FILE__, __LINE__, mysql_error(proxysql_admin));
+		return -1;
+	}
+
+	// Setup the interval of how often new status entries are created
+	uint16_t new_stats_interval_sec = 2;
+
+	// @TODO: run command to set the interval
+
+	// If on a fresh install, wait long enough for the first run of stats to be created
+	// The lookup table will be empty until the first run!
+	sleep(new_stats_interval_sec * 2);
+
+    // Test 1: Lookup table has at least 50 rows
+	int64_t lookup_row_count = 0;
+    MYSQL_QUERY(proxysql_admin, "SELECT COUNT(*) FROM history_mysql_status_variables_lookup");
+	MYSQL_RES* result = mysql_store_result(proxysql_admin);
+	MYSQL_ROW row = mysql_fetch_row(result);
+
+	if (row[0]) 
+		lookup_row_count = strtoll(row[0], nullptr, 10);
+	
+	mysql_free_result(result);
+
+	ok(
+		lookup_row_count >= 50,
+		"Lookup table 'history_mysql_status_variables_lookup' has at least 50 rows. %lu rows found.",
+		lookup_row_count
+	);
+
+	// Test 2: Distinct timestamps in 
+	int64_t distinct_timestamp_count = 0; 
+	MYSQL_QUERY(proxysql_admin, "SELECT COUNT(DISTINCT(timestamp)) FROM history_mysql_status_variables");
+	result = mysql_store_result(proxysql_admin);
+	row = mysql_fetch_row(result);
+
+	if (row[0]) 
+		distinct_timestamp_count = strtoll(row[0], nullptr, 10);
+
+	mysql_free_result(result);
+
+	ok(
```

Comment:
For avoiding timing issues, you should wait at least `new_stats_interval_sec + 1` again before this second test executes, in the CI, it wont fail at this point, but in a freshly start `ProxySQL` and immediately launching this test. This _could_ not hold.